FastSimulation/Tracking/plugins/SimTrackIdProducer.cc

```diff
@@ -51,18 +51,20 @@ SimTrackIdProducer::produce(edm::Event& e, const edm::EventSetup& es)
    
   reco::TrackCollection::const_iterator aTrack = trackCollection->begin();
   reco::TrackCollection::const_iterator lastTrack = trackCollection->end();
-  bool index = 0;
   
-  for ( ; aTrack!=lastTrack; ++aTrack,++index ) {    
-     int SimTrackId = -1;
-    for( trackingRecHit_iterator hit = aTrack->recHitsBegin(); hit != aTrack->recHitsEnd(); ++ hit ) {
-      //   const SiTrackerGSMatchedRecHit2D * rechit = (const SiTrackerGSMatchedRecHit2D*) (hit->get());
-      const SiTrackerGSMatchedRecHit2D * rechit = (const SiTrackerGSMatchedRecHit2D*) (*hit);
-      SimTrackId = rechit->simtrackId();
-      break;
-    }	
-     SimTrackIds->push_back(SimTrackId); 
-    
+  for ( ; aTrack!=lastTrack; ++aTrack)
+  {
+    if((aTrack->quality(aTrack->qualityByName("highPurity")))&&(aTrack->chi2()<9)){
```

Comment:
@shchauha 

Please read the required track quality and the required chi2 from the cfg file